packetery/src/Packetery/class-checkout.php

```diff
@@ -0,0 +1,236 @@
+	 *
+	 * @var array[]
+	 */
+	public static $pickup_point_attrs = array(
+		'id'                   => array(
+			'name'     => 'point_id',
+			'required' => true,
+		),
+		'name'                 => array(
+			'name'     => 'point_name',
+			'required' => true,
+		),
+		'city'                 => array(
+			'name'     => 'point_city',
+			'required' => true,
+		),
+		'zip'                  => array(
+			'name'     => 'point_zip',
+			'required' => true,
+		),
+		'street'               => array(
+			'name'     => 'point_street',
+			'required' => true,
+		),
+		'pickupPointType'      => array(
+			'name'     => 'point_type',
+			'required' => true,
+		),
+		'carrierId'            => array(
+			'name'     => 'carrier_id',
+			'required' => false,
+		),
+		'carrierPickupPointId' => array(
+			'name'     => 'point_carrier_id',
+			'required' => false,
+		),
+		'url'                  => array(
+			'name'     => 'point_url',
+			'required' => true,
+		),
+	);
+
+	/**
+	 * Latte engine
+	 *
+	 * @var \Latte\Engine
+	 */
+	private $latte_engine;
+
+	/**
+	 * Latte engine
+	 *
+	 * @param \Latte\Engine $latte_engine Latte engine.
+	 */
+	public function __construct( \Latte\Engine $latte_engine ) {
+		$this->latte_engine = $latte_engine;
+	}
+
+	/**
+	 * Check if order is bound with Packeta
+	 *
+	 * @return bool
+	 */
+	public static function is_packetery_order(): bool {
+		$chosen_methods = WC()->session->get( 'chosen_shipping_methods' );
+		foreach ( $chosen_methods as $method ) {
+			if ( strpos( $method, 'packetery_shipping_method' ) !== false ) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Renders widget button and information about chosen pickup point
+	 */
+	public function render_widget_button(): void {
+		if ( self::is_packetery_order() ) {
+			$language = substr( get_locale(), 0, 2 );
+
+			$country = strtolower( WC()->customer->get_shipping_country() );
+			if ( ! $country ) {
+				$country = strtolower( WC()->customer->get_billing_country() );
+			}
+			if ( ! $country ) {
+				$this->latte_engine->render( PACKETERY_PLUGIN_DIR . '/template/checkout/error_country.latte' );
+
+				return;
+			}
+
+			$weight = WC()->cart->cart_contents_weight;
+			$weight = wc_get_weight( $weight, 'kg' );
+
+			$this->latte_engine->render(
+				PACKETERY_PLUGIN_DIR . '/template/checkout/widget_button.latte',
+				array(
+					'language' => $language,
+					'country'  => $country,
+					'weight'   => $weight,
+				)
+			);
+		}
+	}
+
+	/**
+	 * Renders main checkout script
+	 */
+	public function render_after_checkout_form(): void {
+		$app_identity = 'woocommerce-' . get_bloginfo( 'version' ) . '-' . WC_VERSION . '-' . Plugin::VERSION;
+
+		$this->latte_engine->render(
+			PACKETERY_PLUGIN_DIR . '/template/checkout/checkout_script.latte',
+			array(
+				'app_identity'       => $app_identity,
+				'pickup_point_attrs' => self::$pickup_point_attrs,
+				'packetery_api_key'  => get_option( 'packetery_api_key' ),
```

Comment:
momentálně to je v get_option( 'packetery_options' )[ 'packetery_api_key' ]

Reply:
V branch 1.0.0 žádné get_option( 'packetery_options' ) není. To tedy dohodneme s Jardou.